### PR TITLE
Rename horizons to auto-sample-conditions

### PR DESCRIPTION
**Pull request by aomarks**
I never liked the name "horizons" (except on the day I chose it I guess). I think a much clearer name is "auto sample conditions".

This changes the `--horizon` flag and the `horizons` JSON config setting (yes... one was singular, the other plural) to `--auto-sample-conditions` and `autoSampleConditions` respectively.

It's backwards compatible, with a warning printed if you use the old style.

Related to https://github.com/Polymer/tachometer/issues/221

### CODE REVIEW

File: src/test/configfile_test.ts [hunk 1]
```diff
@@ -768,6 +768,22 @@ suite('config', () => {
           'config.benchmarks[0].packageVersions requires property "label"'
         );
       });
+
+      test('Error to use both horizons and autoSampleConditions', async () => {
```
Comment (AndrewJakubowicz) [inline src/test/configfile_test.ts:772]: Nice!

**Review by AndrewJakubowicz — APPROVED** [review]
I didn't manually check anything, but the documentation and naming changes look fantastic. They're definitely much clearer. Great test updates as well!

Only tiny optional nits.

🎉

**Review by justinfagnani — APPROVED** [review]

**Comment by Eldeloro1** [comment]
> I never liked the name "horizons" (except on the day I chose it I guess). I think a much clearer name is "auto sample conditions".
> 
> This changes the `--horizon` flag and the `horizons` JSON config setting (yes... one was singular, the other plural) to `--auto-sample-conditions` and `autoSampleConditions` respectively.
> 
> It's backwards compatible, with a warning printed if you use the old style.
> 
> Related to #221

File: README.md [hunk 2]
```diff
@@ -130,31 +130,34 @@ You can change this duration with the `--timeout` flag or the `timeout` JSON
...
-
-In the following example, we have set `--horizon=10%`, meaning we are interested
+### Auto sample conditions
+
+You can also configure which statistical conditions tachometer should check for
+when deciding when to stop auto sampling by configuring _auto sample
+conditions_. This is controlled by the `--auto-sample-conditions` flag or the
+`autoSampleConditions` JSON config option.
+
+An auto sample condition condition can be thought of as a point of interest on
```
[the remark this answers is not in the document]
Comment (Eldeloro1) [reply README.md]: > Nit: `condition` intended twice?

**Pull request merged**